### PR TITLE
rxspencer: add recipe for v3.9.0

### PR DESCRIPTION
**Pull request by nehaljwani**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/rxspencer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    - test -f "${PREFIX}/include/rxspencer/regex.h"                  # [unix]
+    - test -f "${PREFIX}/lib/cmake/rxspencer/RXSpencerConfig.cmake"  # [unix]
+    - test -f "${PREFIX}/lib/librxspencer${SHLIB_EXT}"               # [unix]
+    - if not exist %LIBRARY_INC%\rxspencer\regex.h exit 1            # [win]
+    - if not exist %LIBRARY_INC%\rxspencer\regex.h exit 1            # [win]
+    - if not exist %LIBRARY_LIB%\rxspencer.lib     exit 1            # [win]
+
+about:
+  home: https://garyhouston.github.io/regex
+  license: Custom + CC0-1.0
```
Comment (nehaljwani) [inline recipes/rxspencer/meta.yaml:32]: @conda-forge/staged-recipes What's the correct expression here for: https://github.com/garyhouston/rxspencer/blob/master/COPYRIGHT

Comment (isuruf) [reply recipes/rxspencer/meta.yaml:32]: https://spdx.org/licenses/Spencer-94.html

Comment (nehaljwani) [reply recipes/rxspencer/meta.yaml]: Thanks!

**Comment by nehaljwani** [comment]
@conda-forge/help-c-cpp This is ready for review!

**Pull request merged**